src/onboarding/Deployment/DeploymentStepsItem.js

```diff
@@ -7,9 +7,61 @@ import {
   TRANSACTION_STATUS_SUCCESS,
   TRANSACTION_STATUS_UPCOMING,
 } from '../../symbols'
+import styled from 'styled-components'
+import { addMinutes } from 'date-fns'
 
-function DeploymentStepsItem({ index, name, status }) {
+function DeploymentStepsItem({ index, name, status, dateStart, gasPrice }) {
   const theme = useTheme()
+  const [remainingTime, setRemainingTime] = useState(0)
+  const [remainingTimeUnit, setRemainingTimeUnit] = useState('')
+  const [completedFraction, setCompletedFraction] = useState(0)
+
+  const updateTime = useCallback(
+    dateEnd => {
+      const now = new Date()
+      if (now > dateEnd) {
+        setRemainingTime(0)
+        setRemainingTimeUnit('sec')
+        setCompletedFraction(1)
+        return
+      }
+      const time = (dateEnd - now) / 1000
+      const unit = time < 60 ? 'sec' : time < 3600 ? 'min' : 'hr'
+      const amount = Math.floor(
+        unit === 'sec' ? time : unit === 'min' ? 1 + time / 60 : 1 + time / 3600
+      )
```

Comment:
These methods still seem in development, but I'm gonna take a look. Thanks for the suggestion.

Reply:
These methods still seem in development, but I'm gonna take a look. Thanks for the suggestion.

Reply:
We may also want to take a look at [`dayjs`](https://github.com/iamkun/dayjs) to see if they have something similar. We're trying to move away from `date-fns`.

Reply:
I haven't found any methods in DayJS that give us what we want. I'm leaving this like this for now.

Reply:
It seems straightforward by enabling `dayjs` [embedded plugin for that](https://github.com/iamkun/dayjs/blob/dev/docs/en/Plugin.md#relativetime), we were talking about this the other day, but the change seems still pending

Reply:
The result differs a bit from the design - and it looks OK, so I think it can work. However, now the transaction status line is prone to occupying two lines:
![Screenshot from 2020-01-02 12-28-29](https://user-images.githubusercontent.com/16065447/71681776-99925780-2d5b-11ea-935a-e9144db0f41c.png)

What say you, @javieralaves?

Reply:
@e18r Green light on my end, this is fine